cloud/cluster/cluster.go

```diff
@@ -0,0 +1,65 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
+}
+
+type NodeUpdateType int
+
+const (
+	NodeAdded NodeUpdateType = iota
+	NodeRemoved
+)
+
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+type Cluster interface {
```

Comment:
Before the refactor the Cluster interface had comments for documentation please don't just delete them. 

The new cluster interface needs comments/docs to reflect context